api/room/schema_query.py

```diff
@@ -18,6 +19,37 @@
 from helpers.calendar.analytics_helper import CommonAnalytics
 
 
+def resolve_booked_rooms_analytics(*args):
+    instance, info, start_date, end_date, criteria, limit = args
+    query = Room.get_query(info)
+    location_id = admin_roles.user_location_for_analytics_view()
+    active_rooms = query.filter(
+        RoomModel.state == "active",
+        RoomModel.location_id == location_id)
+    booked_rooms = get_most_and_least_booked_rooms(
+        instance, active_rooms, start_date, end_date, limit, criteria)
+    booked_rooms_statistics = Analytics(
+        analytics=booked_rooms
+    )
+    return booked_rooms_statistics
+
+
+def get_most_and_least_booked_rooms(*args):
+    instance, active_rooms, start_date, end_date, criteria, limit = args
+    if limit and criteria == "least_booked":
+        limit = (-limit)
```

Comment:
Is this assignment necessary here, can we just pass `-limit` directly to the function below?

Reply:
passing `-limit` directly raised an error because of the `-` that why it was assigned before being used.